### DISCUSSION TITLE
Open database (.db) file without schema

### DISCUSSION THREAD

**Discussion by linhttbk97** [Nozbe/WatermelonDB#1701] [Q&A] [answered]
How can I open Database from database file without creating model classes and using raw sql queries to insert, update and also delete data

**Comment by scarlac** [comment, marked as answer]
At that point, you may want to consider a separate SQLite library, as you'd not be using any of the WMDB features at all. Unfortunately, SQLite functionality isn't fully exposed like you want in WMDB, although I'd also like to see these features. One challenge is that WMDB does caching, so if you want to execute arbitrary SQL, it'll get complicated.

However, there is a way. You can open the DB and define a dummy schema with no migrations and no tables (I think?) and then use the "unsafe" functionality that's not officially documented, to execute queries directly. However, I don't really recommend doing that.

**Reply by linhttbk97** [reply]
I need reactive feature like native sqlite library and i don't find any sql library to do that but WMDB. 
I tried to create database and schema but wm required some fields like id, _status and _changed. So the migration steps are really complicated. If my database don't have any id, _status or _changed fields, I got an exception "TypeError: Cannot read property 'id' of null"
